src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1546,16 +1549,38 @@
        new_value, old_value, message, file, line, containment_path, containing_class, corrective_change
      from resource_events"]
    (fn [rows]
-     (reduce (fn [hashes-seen row]
-               (conj hashes-seen
-                     (let [hash-str (hash/resource-event-identity-pkey row)]
-                       (when-not (hashes-seen hash-str)
-                         (jdbc/insert-multi! "resource_events_transform"
-                                             (list (assoc row :event_hash
-                                                          (sutils/munge-hash-for-storage hash-str)))))
-                       hash-str)))
-             #{}
-             rows)))
+     (let [old-cols '(:report_id :certname_id :status :timestamp :resource_type
+                      :resource_title :property :new_value :old_value :message
+                      :file :line :containment_path :containing_class
+                      :corrective_change)
+           new-cols (conj old-cols :event_hash)
+           update-row (apply juxt (conj old-cols
+                                        (comp sutils/munge-hash-for-storage
+                                              hash/resource-event-identity-pkey)))
+           ;; Retrieve event_hash string from PGobject produced by sutils
+           row->id (fn [row] (-> row first .getValue))
+           insert->hash (fn [batch]
+                          (when-not (empty? batch)
```

Comment:
If you decide to change the above, perhaps while you're here `(when (seq? batch) ...)`.

Reply:
Makes sense, flipped it.